lib/modules/dosomething/dosomething_api/resources/campaign_resource.inc

```diff
@@ -327,14 +327,14 @@ function _campaign_resource_reportback($nid, $values) {
     unset($values['northstar_id']);
   }
 
-  // If we pass either a Northstar ID or Drupal UID to `uid`, convert it to a real UID.
-  $uid = dosomething_user_convert_to_legacy_id($values['uid']);
+  // If a Drupal UID is passed to `uid`, convert it to the user's Northstar ID.
+  $uid = dosomething_user_convert_to_northstar_id($values['uid']);
 
   if (empty($uid)) {
     return services_error('Cannot create reportback without a `northstar_id` or `uid`.');
   }
 
-  $user = user_load($uid);
+  $user = dosomething_user_get_user_by_northstar_id($uid);
```

Comment:
Unfortunately [this function](https://github.com/DoSomething/phoenix/blob/21792f5658e5542dde85e9e5bd2f7fbcb24022f9/lib/modules/dosomething/dosomething_user/dosomething_user.module#L971-L989) relies on the local `field_data_field_northstar_id` table, which won't be populated if the user doesn't have a Drupal ID. We want to make sure that this whole endpoint works okay without ever touching the local users table.